static/scripts/pay.ts

```diff
@@ -30,18 +39,29 @@ const withdraw = async (signer: JsonRpcSigner, txData: TxType, predefined: strin
 };
 
 const fetchTreasury = async (): Promise<{ balance: number; allowance: number }> => {
-  const tokenAddress = txData.permit.permitted.token;
-  const tokenContract = new ethers.Contract(tokenAddress, daiAbi, new ethers.providers.Web3Provider((window as any).ethereum));
-  const balance = await tokenContract.balanceOf(txData.owner);
-  const allowance = await tokenContract.allowance(txData.owner, permit2Address);
-  return { balance, allowance };
+  try {
+    const provider = new ethers.providers.Web3Provider((window).ethereum)
+
+    const tokenAddress = txData.permit.permitted.token;
+    const tokenContract = new ethers.Contract(tokenAddress, daiAbi, provider);
+    const balance = await tokenContract.balanceOf(txData.owner);
+    const allowance = await tokenContract.allowance(txData.owner, permit2Address);
+    return { balance, allowance };
+  } catch (error: any) {
+    if(error?.message?.includes("missing provider")) {
+      alert("Error: Please use a web3 enabled browser.");
+    } else {
+      alert("Error: Please connect your wallet.");
+    }
+    return { balance: -1, allowance: -1 };
+  }
 };
 
 const toggleStatus = async (balance: number, allowance: number) => {
   const trBalance = document.querySelector(".tr-balance") as Element;
   const trAllowance = document.querySelector(".tr-allowance") as Element;
-  trBalance.textContent = `$${ethers.utils.formatUnits(balance, 18)}`;
-  trAllowance.textContent = `$${ethers.utils.formatUnits(allowance, 18)}`;
+  trBalance.textContent = balance > 0 ? `$${ethers.utils.formatUnits(balance, 18)}` : 'N/A';
+  trAllowance.textContent = balance > 0 ? `$${ethers.utils.formatUnits(allowance, 18)}` : 'N/A';
```

Comment:
```suggestion
  trBalance.textContent = balance ? `$${ethers.utils.formatUnits(balance, 18)}` : 'N/A';
  trAllowance.textContent = balance ? `$${ethers.utils.formatUnits(allowance, 18)}` : 'N/A';
```

Shouldn't that do the same thing?

Reply:
Yes, it should.. Thanks

Reply:
Now I'm also wondering how USDT and USDC render with `ethers.utils.formatUnits(allowance, 18)` but perhaps that can be a different bounty.

Reply:
> trBalance.textContent = balance ? `$${ethers.utils.formatUnits(balance, 18)}` : 'N/A';

Right, USDT and USDC both have 6 decimals.. DAI has 18

Reply:
> Now I'm also wondering how USDT and USDC render with `ethers.utils.formatUnits(allowance, 18)` but perhaps that can be a different bounty.

I have handled the new issue here